### PR TITLE
Add ImageProjectiveTransform XLA kernel

### PR DESCRIPTION
**Pull request by WindQAQ**
As per comment https://github.com/tensorflow/tensorflow/pull/41365#issuecomment-658818273, cc @tanzhenyu for visibility.

### CODE REVIEW

File: tensorflow/compiler/tf2xla/kernels/image_projective_transform_op.cc [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+    OP_REQUIRES(ctx,
+                output_shape.dim_size(0) > 0 && output_shape.dim_size(1) > 0,
+                errors::InvalidArgument("output dimensions must be positive"));
+
+    output_shape.InsertDim(0, images_shape.dim_size(0));
+    output_shape.InsertDim(3, images_shape.dim_size(3));
+
+    xla::XlaOp fill_value;
+    if (ctx->num_inputs() >= kNumInputsV3) {
```
Comment (tanzhenyu) [inline tensorflow/compiler/tf2xla/kernels/image_projective_transform_op.cc:127]: Do you need this? There is no ImageProjectiveTransformV2 in xla op I think?

**Review by tanzhenyu — COMMENTED** [review]
Thanks for the PR!

File: tensorflow/compiler/tf2xla/kernels/image_projective_transform_op.cc (hunk 1 above)
Comment (WindQAQ) [reply tensorflow/compiler/tf2xla/kernels/image_projective_transform_op.cc:127]: I think it's even unnecessary to register V2 XLA op as XLA kernel appears after V3 commit?

**Comment by WindQAQ** [comment]
@tanzhenyu I think we need an XLA expert to review this. Can you find one? Also, I find the speed does not improve (or even slower) for XLA ops. So maybe there might be some improvement for my handcrafted kernel. Thank you!

**Comment by tanzhenyu** [comment]
From XLA discussion: seems this is not desired given the gathering logic would be expensive.

**Comment by gbaned** [comment]
@WindQAQ  Can you please check @tanzhenyu's comments and keep us posted ? Thanks!

**Comment by WindQAQ** [comment]
@tanzhenyu Do I have to do something specific on the implementation? Thank you!

**Comment by tanzhenyu** [comment]
> @tanzhenyu Do I have to do something specific on the implementation? Thank you!

We can close this if you don't mind. The main reason is that gather takes too much cycles.

**Comment by WindQAQ** [comment]
Yep, sure. As it doesn't improve the performance, let me close it.

**Pull request closed without merging**